finality-aleph/src/data_io/data_provider.rs

```diff
@@ -307,20 +307,21 @@ struct DataProvider<B: BlockT> {
 //    at most MAX_DATA_BRANCH_LEN.
 #[async_trait]
 impl<B: BlockT> aleph_bft::DataProvider<AlephData<B>> for DataProvider<B> {
-    async fn get_data(&mut self) -> AlephData<B> {
-        let data = (*self.data_to_propose.lock().await).clone();
+    async fn get_data(&mut self) -> Option<AlephData<B>> {
+        let data_to_propose = (*self.data_to_propose.lock().await).take();
```

Comment:
This change I like, although I'm sure it would be controversial if timorl was around :P

Anyway, could you perhaps add some little comment to `ChainTracker` that the other thread is "consuming" (and leaving `None`) from `data_to_propose`. I imagine one could sometime in the future make a wrong assumption that `get_data` does not modify `data_to_propose`.

Reply:
Done.